examples/decoding/cohy_decomposition.py

```diff
@@ -0,0 +1,532 @@
+]
+
+########################################################################################
+# There are two equivalent options for fitting and transforming the same data: 1)
+# passing the data to the :meth:`~mne_connectivity.decoding.MIC.fit` and
+# :meth:`~mne_connectivity.decoding.MIC.transform` methods sequentially; or 2) using the
+# combined :meth:`~mne_connectivity.decoding.MIC.fit_transform` method.
+#
+# We use the latter approach below, fitting the filters to the 15-20 Hz band and using
+# the ``"imcoh"`` method in the call to the ``spectral_connectivity_...()`` functions.
+# Plotting the results, we see a peak in connectivity at 15-20 Hz.
+
+# %%
+
+mic = MIC(info=epochs.info, fmin=FMIN, fmax=FMAX, indices=(seeds, targets), rank=(3, 3))
+
+start = time.time()
+epochs_transformed = mic.fit_transform(epochs.get_data())
+
+con_mic_class = spectral_connectivity_epochs(
+    epochs_transformed,
+    method="imcoh",
+    indices=mic.get_transformed_indices(),
+    fmin=5,
+    fmax=30,
+    sfreq=epochs.info["sfreq"],
+)
+class_duration = time.time() - start
+
+ax = plt.subplot(111)
+ax.plot(
+    con_mic_class.freqs,
+    np.abs(con_mic_class.get_data()[0]),
+    color=plt.rcParams["axes.prop_cycle"].by_key()["color"][2],
+    label="Decomposition class",
+)
+ax.axvspan(FMIN, FMAX, color="grey", alpha=0.2, label="Fitted freq. band")
+ax.set_xlabel("Frequency (Hz)")
+ax.set_ylabel("Connectivity (A.U.)")
+ax.set_title("MIC")
+plt.legend()
+plt.show()
+
+########################################################################################
+# For comparison, we can also use the standard approach of the
+# ``spectral_connectivity_...()`` functions, which shows a very similar connectivity
+# profile in the 15-20 Hz frequency range. Bivariate coherence is again shown to
+# demonstrate the signal-to-noise enhancements the multivariate approach offers.
+
+# %%
+
+start = time.time()
+con_mic_func = spectral_connectivity_epochs(
+    epochs.get_data(),
+    method="mic",
+    indices=([seeds], [targets]),
+    fmin=5,
+    fmax=30,
+    sfreq=epochs.info["sfreq"],
+    rank=([3], [3]),
+)
+func_duration = time.time() - start
+
+con_imcoh = spectral_connectivity_epochs(
+    epochs.get_data(),
+    method="imcoh",
+    indices=seed_target_indices(seeds, targets),
+    fmin=5,
+    fmax=30,
+    sfreq=epochs.info["sfreq"],
+    rank=([3], [3]),
+)
+
+ax = plt.subplot(111)
+ax.plot(
+    con_mic_func.freqs,
+    np.abs(con_mic_func.get_data()[0]),
+    label="MIC (standard\nfunction)",
+)
+ax.plot(
+    con_imcoh.freqs,
+    np.mean(np.abs(con_imcoh.get_data()), axis=0),
+    label="ImCoh",
+)
+ax.plot(
+    con_mic_class.freqs,
+    np.abs(con_mic_class.get_data()[0]),
+    label="MIC (decomposition\nclass)",
+)
+ax.axvspan(FMIN, FMAX, color="grey", alpha=0.2, label="Fitted freq. band")
+ax.set_xlabel("Frequency (Hz)")
+ax.set_ylabel("Connectivity (A.U.)")
+ax.set_title("MIC")
+plt.legend()
+plt.show()
```

Comment:
Perhaps explain once more that the difference between `MIC (standard function)` and `MIC (decomposition class)` is that the letter will fit a single filter for the entire band, whereas the standard function will not. Hence the estimated coherence differs.

Reply:
Well, you do that in the Limitations section right after this. Hrm. Perhaps it could be annotated in the plot somehow? I just got confused a little why there would be any difference at all between using the function or the decomposition class. A little reminder would be nice.

Reply:
Good point, I can maybe introduce this earlier.